### DISCUSSION TITLE
Does the `--grammar` option work in `llama-server`?

### DISCUSSION THREAD

**Discussion by jukofyork** [ggml-org/llama.cpp#10592] [Q&A] [answered]
Just wondering if it works now as last time I tried it seemed to never be used, and I see on reddit people recommending it for use with the new `QwQ` model that keeps reverting to Chinese?

**Comment by ggerganov** [comment, marked as answer]
Yes, it should work: https://github.com/ggerganov/llama.cpp/pull/10612

**Comment by jukofyork** [comment]
Thanks, I'll add the two examples I failed to get working in the PR thread - it was shortly after the unifying of the CLI parameters for `server` so I thought it was likely just an unimplemented option back then, but it's possible the grammar (or even the idea of blocking the " CJK" glyphs) was bad (I tried using command line and in a file IIRC).
